### ISSUE TITLE
Urgent bug - Categories are being displayed before products on all shop archives

### ISSUE THREAD

**Issue by JiveDig** [copyblogger/genesis-connect-woocommerce#15]
I have 3 products in the Mountain Bike category. When I go to the Mountain Bike category archive I see this: 

![screen shot 2018-01-30 at 3 26 07 pm](https://user-images.githubusercontent.com/540990/35589473-f8bed1ba-05d1-11e8-985c-2eaa14685fa9.png)

That is the entire category loop. It displays on the shop and on all product cat/tag archives.

In `genesiswooc_content_product()` there is `woocommerce_product_subcategories()`. I think there needs to be some checks like:

```
$display_type = woocommerce_get_loop_display_mode();
If displaying categories, append to the loop.
if ( 'subcategories' === $display_type || 'both' === $display_type ) {
	woocommerce_product_subcategories();
}
```
It works, but not fully tested. IMO this is a breaking bug and needs to get fixed ASAP.

FWIW, I'm pretty sure something changed in Woo 3.3.0 that caused this. I can't recreate the bug pre-3.3, but I can see it on various G child themes, and G itself, on different servers/installs.

**Comment by dreamwhisper** [comment]
@nathanrice  We have a couple of tickets in desk about this and I've replicated on our test demos.

**Comment by SSSV** [comment]
I am experiencing the same thing. This is with G-Connect activated...

![image 2018-01-30 at 12 32 02 pm](https://user-images.githubusercontent.com/10344534/35590312-4bc79804-05bb-11e8-9342-52d22c70786e.png)

With G-Connect deactivated, a random image is being pulled in to the Woo product archive. This image is not the image associated with that product category so I'm not sure where this is coming from.

![image 2018-01-30 at 12 27 21 pm](https://user-images.githubusercontent.com/10344534/35590343-6da4327a-05bb-11e8-8a19-9ab571b0d6c9.png)

**Comment by nathanrice** [comment]
Can someone who is having the issue take a look at 376a864d89d82efc166cd26088864532a4390475 and maybe pull down the changed file and swap it into your plugin to see if that clears things up for you?

**Comment by dreamwhisper** [comment]
@nathanrice It's working on the test links I sent you in Slack. 

If anyone has any oddities even with the file update, resave `wp-admin/admin.php?page=wc-settings&tab=products`

**Comment by timothyjensen** [comment]
I tested the [WooCommerce PR](https://github.com/woocommerce/woocommerce/pull/18676) that addresses this very issue, and it restores the product archive is back to normal (no product categories at the top of the archive).

Nathan's latest commit also fixes the product archive, which means I don't need to wait for WooCommerce to release 3.3.1 👍.

**Comment by claudiulodro** [comment]
The culprit is a behavior change in `woocommerce_product_subcategories`. We'll have a backwards-compatibility patch in WooCommerce 3.3.1(https://github.com/woocommerce/woocommerce/pull/18676), but updating the template file to match the WooCommerce archive-product.php template in 3.3 will also fix the issue.

**Comment by Frenchomatic** [comment]
@nathanrice  Looking good.  WP 4.9.2 + Genesis 2.5.3  + Infinity Pro theme.  You may need to go into the customzier and reset the number of products in a row again but apart from that your patch seems to work for me.

**Comment by nickcernis** [comment]
This patch is also working in my tests.

**Comment by kardingdeira** [comment]
Hi Is these working for any templates?

**Comment by nickcernis** [comment]
@kardingdeira If you're using a Genesis child theme with Genesis Connect and seeing categories on your shop page, there's a manual template fix for this issue you can use here: https://wordpress.org/support/topic/categories-showing-on-shop-page-after-wc-3-3/#post-9921116

If you need further help with, you're welcome to start a new topic in the WordPress plugin forum, submit a ticket at https://my.studiopress.com/help/, or email `support [at] studiopress [dot] com`.

**Comment by kardingdeira** [comment]
thank you, sorry I am not using a Genesis theme, but is other Template

**Comment by nickcernis** [comment]
@kardingdeira No problem! If you're experiencing this with another theme it's best to contact the theme developer. You are welcome to link to this page in case it helps to solve their own theme issue faster.

**Comment by kardingdeira** [comment]
Great Thank you

Billy Booc




M. +971 52 2436001

EM. saigon@billyboocarts.com

W. http://billybooc.com





 

    On Wednesday, January 31, 2018, 4:17:41 PM GMT+4, Nick Cernis <notifications@github.com> wrote:  
 
 
@kardingdeira No problem! If you're experiencing this with another theme it's best to contact the theme developer. You are welcome to link to this page in case it helps to solve their own theme issue faster.

**Comment by nickcernis** [comment]
Confirmed this is fixed in production with the 0.9.10 release.
